core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulseDataCollector.java

```diff
@@ -215,27 +213,74 @@ private void addRequest(RequestData requestTelemetry) {
             return;
         }
 
-        counters.requestsAndDurations.addAndGet(Counters.encodeCountAndDuration(1, toMilliseconds(requestTelemetry.getDuration())));
+        counters.requestsAndDurations.addAndGet(Counters.encodeCountAndDuration(1, parseDurationToMillis(requestTelemetry.getDuration())));
         if (!requestTelemetry.isSuccess()) {
             counters.unsuccessfulRequests.incrementAndGet();
         }
     }
 
-    // FIXME (trask) move live metrics request capture to OpenTelemetry layer so don't have to parse String duration?
-    private static long toMilliseconds(String duration) {
+    // TODO (trask) move live metrics request capture to OpenTelemetry layer so don't have to parse String duration
+    // visible for testing
+    static long parseDurationToMillis(String duration) {
         // format is DD.HH:MM:SS.MMMMMM
-        StringTokenizer tokenizer = new StringTokenizer(duration, ".:");
-        int days = Integer.parseInt(tokenizer.nextToken());
-        int hours = Integer.parseInt(tokenizer.nextToken());
-        int minutes = Integer.parseInt(tokenizer.nextToken());
-        int seconds = Integer.parseInt(tokenizer.nextToken());
-        int microseconds = Integer.parseInt(tokenizer.nextToken());
-
-        return Duration.ofDays(days)
-                .plusHours(hours)
-                .plusMinutes(minutes)
-                .plusSeconds(seconds)
-                .plusNanos(microseconds * 1000L)
-                .toMillis();
+        return startingAtDaysOrHours(duration);
+    }
+
+    private static long startingAtDaysOrHours(String duration) {
```

Comment:
where is the doc for this? is breeze so picky about duration? this is new to me :)

Reply:
just reverse engineered from 3.1.1 behavior